VMBackup/main/HttpUtil.py

```diff
@@ -33,23 +33,26 @@
 
 class HttpUtil(object):
     """description of class"""
-    def __init__(self, hutil):
-        Config = None
-        self.proxyHost = None
-        self.proxyPort = None
-        try:
-            waagent.MyDistro = waagent.GetMyDistro()
-            Config = waagent.ConfigurationProvider(None)
-        except Exception as e:
-            errorMsg = "Failed to construct ConfigurationProvider, which may due to the old wala code with error: %s, stack trace: %s" % (str(e), traceback.format_exc())
-            hutil.log(errorMsg)
-            Config = None
-        self.logger = hutil
-        if Config != None:
-            self.proxyHost = Config.get("HttpProxy.Host")
-            self.proxyPort = Config.get("HttpProxy.Port")
-
-        self.tmpFile = './tmp_file_FD76C85E-406F-4CFA-8EB0-CF18B123365C'
+    __instance = None
+    def __new__(cls, hutil):
+        if(cls.__instance is None):
+            hutil.log("Creating HttpUtil")
+            cls.__instance = super(HttpUtil, cls).__new__(cls)
+            try:
+                waagent.MyDistro = waagent.GetMyDistro()
+                Config = waagent.ConfigurationProvider(None)
```

Comment:
On a customer VMs, it seems above waagent APIs were getting stuck during freeze. So by making it singleton, these will not happen during freeze. With this private-fix, backup of customer's 7-8 VMs have started succeeding which were consistently failing for months.